core/src/main/java/org/apache/iceberg/rest/requests/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * A REST request to create a namespace, with an optional set of properties.
+ * The server might also add properties, such as `last_modified_time` etc.
+ */
+public class CreateNamespaceRequest {
+
+  private Namespace namespace;
+  private Map<String, String> properties;
+
+  public CreateNamespaceRequest() {
+    // Required for serialization
+  }
+
+  private CreateNamespaceRequest(Namespace namespace, Map<String, String> properties) {
+    this.namespace = namespace;
+    this.properties = properties;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public Namespace namespace() {
+    return namespace;
+  }
+
+  public Map<String, String> properties() {
+    return ImmutableMap.copyOf(properties);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("namespace", namespace)
+        .add("properties", properties())
+        .toString();
+  }
+
+  public static class Builder {
+    private final ImmutableList.Builder<String> namespaceBuilder;
+    private final ImmutableMap.Builder<String, String> propertiesBuilder;
+
+    private Builder() {
+      this.namespaceBuilder = ImmutableList.builder();
+      this.propertiesBuilder = ImmutableMap.builder();
+    }
+
+    public Builder withNamespace(Namespace namespace) {
+      this.namespaceBuilder.add(namespace.levels());
+      return this;
+    }
+
+    public Builder withProperties(Map<String, String> properties) {
```

Comment:
I don't think it's clear from the name that this is not replacing the set of properties. I think a verb like `set` would fix this. For example, `setProperties(map)` implies to me that you're adding all the new properties and not removing anything.

Reply:
Updated (and added a check for `null` on the input as well as I see people do that sometimes in tests etc).